testsuite/MDAnalysisTests/auxiliary/base.py

```diff
@@ -357,7 +357,33 @@ def test_load_from_description(self, reader):
         new = mda.auxiliary.core.auxreader(**description)
         assert new == reader, "AuxReader reloaded from description does not match"
 
-    
+    def test_step_to_frame(self, ref, reader):
+
+        ts = mda.coordinates.base.Timestep(0, dt=1)
+
+        assert reader.step_to_frame(-1, ts) is None
+        assert reader.step_to_frame(reader.n_steps, ts) is None
+        assert reader.step_to_frame(0, ts) == 0
+        
+        frame, time_diff = reader.step_to_frame(1, ts, return_time_diff=True)
+
+        assert frame == 1
+        assert time_diff == pytest.approx(0)
+
+        # Timestep is 0.2 longer than auxiliary data
```

Comment:
Try and split tests so each test does one thing. This ts is independent of the previous

Reply:
Good point. I created `test_step_to_frame_out_of_bound`, `test_step_to_frame_no_time_diff` and `test_step_to_frame_time_diff`.